src/Mailgun/Connection/RestClient.php

```diff
@@ -349,6 +349,14 @@ public function setApiVersion($apiVersion)
     }
 
     /**
+     * @return string
+     */
+    public function getApiVersion()
```

Comment:
I would be happy to remove the setter as well =)

I have an idea that the RestClient should only care about building a request and handle the response. That includes adding headers and exception handling. 

To increase flexibility of the RestClient we should not allow it to know about API version or the URLs. That should be handled in the MailGun class.